src/main/java/io/cdap/plugin/util/FQNGenerator.java

```diff
@@ -0,0 +1,56 @@
+package io.cdap.plugin.util;
+
+import io.opentracing.contrib.jdbc.ConnectionInfo;
+import io.opentracing.contrib.jdbc.parser.URLParser;
+
+/**
+ * Generate FQN from DB URL Connection
+ */
+public class FQNGenerator {
+
+  private static final String POSTGRESQL_TAG = "postgresql";
+  private static final String POSTGRESQL_DEFAULT_SCHEMA = "public";
+
+  public static String constructFQN(String url, String tableName) {
+    // dbtype, host, port, db from the connection string
+    // table is the reference name
+    ConnectionInfo connectionInfo = URLParser.parse(url);
+    // DB type as set by library after extraction
+    if (POSTGRESQL_TAG.equals(connectionInfo.getDbType())) {
+      // FQN for Postgresql
+      return String.format("%s://%s/%s.%s.%s", connectionInfo.getDbType(), connectionInfo.getDbPeer(),
+                           connectionInfo.getDbInstance(), getPostgresqlSchema(url), tableName);
+    } else {
+      // FQN for MySQL, Oracle, SQLServer
+      return String.format("%s://%s/%s.%s", connectionInfo.getDbType(), connectionInfo.getDbPeer(),
+                           connectionInfo.getDbInstance(), tableName);
+    }
+  }
+
+  private static String getPostgresqlSchema(String url) {
```

Comment:
this is very complicated logic, I am not sure why postgres is a special case here. How do we emit fqn for generic db which also takes a jdbc string